File: 01-Arrays-and-Hashing/1. Two Sum/1. Two Sum.cpp
1class Solution {
2public:
3    vector<int> twoSum(vector<int>& nums, int target) {
4
5        unordered_map<int,int> hash;
6        for(int i=0;i<nums.size();i++){
7            int complement=target-nums[i];
8            if(hash.count(complement))return {hash[complement],i};
9            else hash[nums[i]]=i;
10        }
11        return {0};
12    
13    }
14};

File: 01-Arrays-and-Hashing/1. Two Sum/1. Two Sum.java
1class Solution {
2    public int[] twoSum(int[] nums, int target) {
3        Map<Integer, Integer> map = new HashMap<>();
4
5        int n = nums.length;
6
7        for(int i=0; i<n; i++){
8         for(int j=i+1; j<n; j++){
9            if(nums[i] + nums[j] == target){
10                return new int[]{i,j};
11            }
12         }
13        }
14
15        return new int[]{};
16    }
17}